packages/react/src/ReactElement.js

```diff
@@ -179,6 +180,16 @@ export function jsx(type, config, maybeKey) {
   let key = null;
   let ref = null;
 
+  // Currently, key can be spread in as a prop. This causes a potential
+  // issue if key is also explicitly declared (ie. <div {...props} key="Hi" />
+  // or <div key="Hi" {...props} /> ). We want to deprecate key spread,
+  // but as an intermediary step, we will use jsxDEV for everything except
+  // <div {...props} key="Hi" />, because we aren't currently able to tell if
+  // key is explicitly declared to be undefined or not.
+  if (maybeKey !== undefined) {
```

Comment:
I like lifting this, but I think it should be right next to the `hasValidKey` check on 197. Especially since this comment applies to both checks. Might be better to lift it as well to be above the `ref` assignment.